res/enkf/es_update.py

```diff
@@ -97,18 +97,43 @@ def analysis_smoother_update(
                 scaling attached. These parameters are updated one at a time.
             """
 
-            module_config = _lib.analysis_module.get_module_config(module)
-            module_data = _lib.analysis_module.get_module_data(module)
+            module_config = analysis_module.get_module_config(module)
+            module_data = analysis_module.get_module_data(module)
+
             if update_data.A is not None:
-                _lib.update.run_analysis_update_without_rowscaling(
-                    module_config, module_data, ens_mask, update_data
-                )
+                if module_config.iterable():
+                    ies.init_update(module_data, ens_mask, update_data.obs_mask)
+                    iteration_nr = module_data.inc_iteration_nr()
```

Comment:
Is there a better way of doing this?

Reply:
yes, and should be included in future refactoring of the ies module